samza-kv/src/main/scala/org/apache/samza/storage/kv/BaseKeyValueStorageEngineFactory.scala

```diff
@@ -83,6 +83,12 @@ trait BaseKeyValueStorageEngineFactory[K, V] extends StorageEngineFactory[K, V]
     val storeFactory = storageConfig.get("factory")
     var storePropertiesBuilder = new StoreProperties.StorePropertiesBuilder()
 
+    val accessLog = if (storageConfig.containsKey("accesslog.status")) {
```

Comment:
Here, storageConfig is not of type StorageConfig and I get the following error if I try to cast it: 
``` java.lang.ClassCastException: org.apache.samza.config.MapConfig cannot be cast to org.apache.samza.config.StorageConfig ```
I've refactored it to use getBoolean(value, default) though